addons/binding/org.openhab.binding.tankerkoenig/src/main/java/org/openhab/binding/tankerkoenig/handler/BridgeHandler.java

```diff
@@ -0,0 +1,382 @@
+                                        break;
+                                    case 6:
+                                        if ((day.contains("Wochendende")) || (day.contains("Sa"))) {
+                                            logger.debug("Found a setting which is valid for today (Saturday).");
+                                            foundIt = true;
+                                        }
+                                        break;
+                                    case 7:
+                                        if ((day.contains("Wochenende")) || (day.contains("So"))) {
+                                            logger.debug("Found a setting which is valid for today (Sunday).");
+                                            foundIt = true;
+                                        }
+                                        break;
+                                }
+                                if (foundIt) {
+                                    start = open;
+                                    ende = close;
+                                    i = o.length;
+                                }
+                            }
+                            i = i + 1;
+                        } while (i < o.length);
+                    }
+                } else {
+                    // no OpeningTimes found, assuming WholeDay open!
+                    foundIt = true;
+                    logger.debug("No OpeningTimes are found, assuming WholeDay.");
+                    // "start" and "ende" are set manually!
+                    start = "00:00";
+                    ende = "23:59";
+                }
+                LocalTime tempopening = LocalTime.parse(start);
+                LocalTime tempclosing = LocalTime.parse(ende);
+                DateTime opening = tempopening.toDateTimeToday();
+                DateTime closing = tempclosing.toDateTimeToday();
+                if (opening.isEqual(closing)) {
+                    closing = opening.plusDays(1);
+                } else {
+                    // Tankerkoenig.de does update the status "open" every 4 minutes
+                    // due to this the status "open" could be sent up to 4 minutes after the published opening time
+                    // therefore the first update is called 4 minutes after opening time!
+                    opening = opening.plusMinutes(4);
+                }
+                if ((now.isAfter(opening) & (now.isBefore(closing)))) {
+                    // logger.debug("Opening: {}, Closing: {}, now: {}", opening, closing, now);
+                    logger.debug("Now is within opening times for today.");
+                    if (sb.toString().equals("")) {
+                        sb.append(tkh.getLocationID());
+                    } else {
+                        sb.append("," + tkh.getLocationID());
+                    }
+                }
+            }
+            return sb.toString();
+        } catch (Exception e) {
```

Comment:
Touché.
The only specific exception wouldbe ParseException, but that is thrown by the method already. Removed Try..Catch.